readthedocs/api/v2/views/footer_views.py

```diff
@@ -216,6 +216,20 @@ def _get_context(self):
         }
         return context
 
+    def is_ad_free_user(self, user):
+        if not settings.USE_PROMOS:
+            return True
+        if user.is_authenticated and hasattr(user, 'gold') and hasattr(user, 'goldonce') and (user.gold.exists() or user.goldonce.exists()):
+            return True
+        return False
+
+    def is_ad_free_project(self, project):
+        if not settings.USE_PROMOS:
+            return True
+        if project and hasattr(project, 'gold_owners') and (project.gold_owners.exists() or project.ad_free):
```

Comment:
The check `hasattr(project, 'gold_owners')` just checks whether that object is present, not whether there actually is a gold owner. This is needed because the gold app may not be installed.

Reply:
Gotcha -- we should probably just check for that explicitly (`if 'donate' in settings.INSTALLED_APPS`), or at least add a comment. It isn't clear that's the logic from the code.

Reply:
When this if's gets complicated on the `or` things, I try to split per each case to avoid confusions, like:

```python
if not settings.USE_PROMOS:
  # commercial site
  return True

if project.ad_free:
  # marked manually as ad free
  return True

if project and hasattr(project, 'gold_owners') and project.gold_owners.exists():
  # project's owner is gold member
  return True

return False
```

You end up with multiple exits where the output is the same, `True`, but it's clearer to read, IMO.

Reply:
After discussing, we're going to check `settings.INSTALLED_APPS`.